src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1870,7 +1870,7 @@ internal static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSt
                 else
                 {
                     var property = sourceMethod.AssociatedSymbol as SourcePropertySymbolBase;
-                    if ((object)property != null && property.IsAutoPropertyWithGetAccessor) // PROTOTYPE: Just IsAutoProperty, or _isAutoPropertyAccessor?
+                    if ((object)property != null && property.IsAutoProperty && property.GetMethod is not null) // PROTOTYPE: Just IsAutoProperty, or _isAutoPropertyAccessor?
```

Comment:
>property.IsAutoProperty && property.GetMethod is not null [](http://example.com/codeflow?start=52&length=57)

If I remember correctly, in one of the comments on the initial PR I indicated that we simply shouldn't be asking property about anything here. An accessor should itself know if its body is supposed to be synthesized. This is the kind of refactoring I had in mind for this place.

Reply:
@AlekseyTs I refactored this a bit, but I ended up that I'll need to ask the property whether it has a get accessor or not, otherwise we'll call `ConstructAutoPropertyAccessorBody` and crash because there is no backing field for a semi-colon setter only property (e.g `public int P { set; }`).

We can prevent the crash by creating a backing field in absence of get accessor, but I don't think we really want to unnecessarily create a backing field, which also (may) affect other parts of API consumers.

What do you think?

Reply:
>  I refactored this a bit, but I ended up that I'll need to ask the property whether it has a get accessor or not

This sounds wrong. If accessor knows whether its body is supposed to be synthesized and that is something that is known at its construction time, then why do we need to ask property anything here?

Reply:
Here is the relevant thread from the initial PR - [https://github.com/dotnet/roslyn/pull/57076/commits/82b632a0c8b57117ff60f8c09c08b3f599b9aae4#r773483332](https://github.com/dotnet/roslyn/pull/57076/commits/82b632a0c8b57117ff60f8c09c08b3f599b9aae4#r773483332)